code/entities/WireGateEntity.cs

```diff
@@ -351,6 +353,92 @@ public void WireInitialize()
 				storedEnt = value;
 			} );
 		}
+		else if ( GateType == "Latch" )
+		{
+			this.RegisterInputHandler( "Value", (float value) => { });
+			this.RegisterInputHandler( "Clk", ( bool value ) =>
+			{
+				if ( value )
+				{
+					this.WireTriggerOutput( "Out", inputs["Value"].asFloat );
+				}
+			} );
+		}
+		else if ( GateType == "D-Latch" )
+		{
+			BulkRegisterInputHandlers( ( float value ) =>
+			{
+				if ( inputs["On"].asBool )
+				{
+					this.WireTriggerOutput( "Out", inputs["Value"].asFloat );
+				}
+			}, new string[] { "Value", "On" } );
+		}
+		else if ( GateType == "Toggle" )
+		{
+			this.RegisterInputHandler( "Toggle", ( bool value ) =>
+			{
+				if ( value )
+				{
+					float newValue = Math.Abs( storedFloat - 1 ) < 0.01 ? 0 : 1;
+					storedFloat = newValue;
+					this.WireTriggerOutput( "Out", newValue );
+				}
+			} );
+		}
+		else if ( GateType == "RAM" )
+		{
+			storedRAM ??= new float[32768];
+			this.RegisterInputHandler( "Address", ( float value ) =>
+			{
+				if ( (int)value >= 0 && (int)value < 32768 )
+				{
+					this.WireTriggerOutput( "Out", storedRAM[(int)value] );
+				}
+			} );
+			this.RegisterInputHandler( "Value", ( float value ) => { } );
+			this.RegisterInputHandler( "Clk", ( bool value ) =>
```

Comment:
How do you feel about naming this input "Write"? Maybe same for the Latch too?
I'm just trying to think how I'd explain Clk to someone who'd never seen it before: "Clk, its short for Clock Signal, which  is an electronics term for a signal that regularly occilates between a high and low level..." vs "Send a value to the Write input and it Writes to the Latch/RAM's internal state"

Reply:
Yeah, I'm fine with that being changed to "write" it's definitely more intuitive about what the signal actually does than what it is currently.